pkg/vendir/fetch/imgpkgbundle/sync.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+package imgpkgbundle
+
+import (
+	"bytes"
+	"fmt"
+	ctlconf "github.com/vmware-tanzu/carvel-vendir/pkg/vendir/config"
+	ctlfetch "github.com/vmware-tanzu/carvel-vendir/pkg/vendir/fetch"
+	"os/exec"
+	"regexp"
+	"strings"
+)
+
+type Sync struct {
+	opts       ctlconf.DirectoryContentsImgpkgBundle
+	refFetcher ctlfetch.RefFetcher
+}
+
+func NewSync(opts ctlconf.DirectoryContentsImgpkgBundle, refFetcher ctlfetch.RefFetcher) *Sync {
+	return &Sync{opts, refFetcher}
+}
+
+var (
+	// Example image ref in imgpkg stdout:
+	//   Pulling bundle 'index.docker.io/dkalinin/consul-helm@sha256:d1cdbd46561a144332f0744302d45f27583fc0d75002cba473d840f46630c9f7'
+	imgpkgPulledImageRef = regexp.MustCompile("(?m)^Pulling bundle '(.+)'$")
```

Comment:
@joaopapereira would be good to figure out a better way to do this instead of parsing imgpkg output. we do have --lock-ouptut but can we somehow grab it without temp file?

Reply:
(not a blocker for this PR imho)